data/data-pipeline/data_pipeline/tests/score/test_output.py

```diff
@@ -203,3 +212,44 @@ def test_donut_hole_addition_to_score_n(final_score_df):
     assert (
         new_donuts > 0
     ), "FYI: The adjacency index is doing nothing. Consider removing it?"
+
+
+def test_data_sources(
+    final_score_df, hud_housing_df, ejscreen_df, cdc_places_df
+):
+    data_sources = {
+        key: value for key, value in locals().items() if key != "final_score_df"
+    }
+
+    for data_source_name, data_source in data_sources.items():
+        final = "_final"
+        df: pd.DataFrame = final_score_df.merge(
+            data_source,
+            on=GEOID_TRACT_FIELD_NAME,
+            indicator="MERGE",
+            suffixes=(final, f"_{data_source_name}"),
+            how="left",
```

Comment:
So i went back and forth on left vs outer, and i landed on left because I was honestly unsure what i should do with right_only tracts in this case --- like, we know some sources have tracts beyond our 2010 final list, so I don't think it's an error if there are right-only tracts. I could log it --- just sorta didn't think anyone would ever see the logs from passing tests tbh.

I'm curious to hear more about what you're thinking about "we could also ensure that if that happens when data is refreshed (e.g., if a source silently moves to 2020 tract boundaries) we would throw an error" --- at the time this test is run, it doesn't know what the previous data looked like for any given source, so I'm not sure on what condition I'd assert an error, but I might  not be thinking that through all the way.